dawn/src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -88,6 +88,16 @@ struct ComputeEditDistance<std::string> {
   }
 };
 
+std::pair<std::string, std::string> dirnameBasename(const std::string& filename) {
+
+  const char sep = '/';
+
+  const auto i = filename.rfind(sep);
```

Comment:
I saw a lot of path manipulation in prs recently. We should consider using std::filesystem.

Reply:
Ah, yes I’m glad you said that. There are different parts of the code all using their own parsing methods. I will change that in this PR before we merge.

Reply:
I used it in this place. We'll have to track down the other places later.